lib/pipeline.js

```diff
@@ -6,14 +6,55 @@ const BaseModel = require('./base');
 const parser = require('screwdriver-config-parser');
 const workflowParser = require('screwdriver-workflow-parser');
 const hoek = require('hoek');
-const { PR_JOB_NAME } = require('screwdriver-data-schema').config.regex;
+const { PR_JOB_NAME, EXTERNAL_TRIGGER } = require('screwdriver-data-schema').config.regex;
 const PAGINATE_PAGE = 1;
 const PAGINATE_COUNT = 50;
 const REGEX_CAPTURING_GROUP = {
     pr: 1, // PR-1
     job: 2 // main or undefined if using legacy
 };
 
+/**
+ * Sync external triggers
+ * Remove the trigger if the job is no longer in yaml, or if the src is no longer in 'requires'
+ * Add the trigger src is in 'requires' and was not in the database yet
+ * @method syncExternalTriggers
+ * @param  {Object}     config              config
+ * @param  {Number}     config.pipelineId   pipelineId belongs to this job
+ * @param  {String}     config.jobName      name of the job
+ * @param  {Array}      config.requiresList requires value of this job
+ * @return {Promise}
+ */
+function syncExternalTriggers(config) {
+    // Lazy load factory dependency to prevent circular dependency issues
+    // https://nodejs.org/api/modules.html#modules_cycles
+    /* eslint-disable global-require */
+    const TriggerFactory = require('./triggerFactory');
+    /* eslint-enable global-require */
+
+    const triggerFactory = TriggerFactory.getInstance();
+    const newSrcList = config.requiresList;
+    const dest = `~sd@${config.pipelineId}:${config.jobName}`;
+    const processed = [];
+
+    // list records that would trigger this job
+    return triggerFactory.list({ params: { dest } })
+    .then((records) => {
+        // if the src is not in the new src list, then remove it
+        const toRemove = records.filter(rec => !newSrcList.includes(rec.src));
+
+        // if the new src is not in the old src list, then create in datastore
```

Comment:
I think you can omit "new" here so it's consistent with the comment above. Otherwise the two src sound like different src to me.

Reply:
Fixed!